.changeset/thirty-beds-smoke.md

```diff
@@ -0,0 +1,8 @@
+---
+"@astrojs/mdx": minor
+"@astrojs/markdown-remark": minor
+---
+
+Handle syntax highlighting using rehype plugins instead of remark plugins. This provides better interoperability with other [rehype plugins](https://github.com/rehypejs/rehype/blob/main/doc/plugins.md#list-of-plugins) that deal with code blocks, in particular with third party syntax highlighting plugins and [`rehype-mermaid`](https://github.com/remcohaszing/rehype-mermaid).
+
+This may break your code if you are using either a remark plugin that relies on nodes of type `html`, or a rehype plugin that depends on nodes of type `raw`. If you are, you should consider using a rehype plugin that deals with the generated `element` nodes instead. You can use [`hast-util-to-html`](https://github.com/syntax-tree/hast-util-to-html) to get a string from a `raw` node, but better is to transform the AST instead of raw HTML strings.
```

Comment:
```suggestion
This may be a breaking change if your are currently using:
- a remark plugin that relies on nodes of type `html`
- a rehype plugin that depends on nodes of type `raw`. 

Please review your rendered code samples carefully, and if necessary, consider using a rehype plugin that deals with the generated `element` nodes instead. You can transform the AST of raw HTML strings, or alternatively use [`hast-util-to-html`](https://github.com/syntax-tree/hast-util-to-html) to get a string from a `raw` node.
```
Just a suggestion because clarity is always good when someone's project code might break because of a change!

So I've called attention to what the problem could be and put the desired solution first in the list (rather than "do this... or better, do this).

Please correct if this is now inaccurate because of moving words around, but I think something that is visually easier to see the breaking change is helpful here!